index.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const Ajv = require('ajv')
+
+function ValidatorSelector () {
+  const validatorPool = new Map()
+  const cache = new Map()
+  cache.put = cache.set
+
+  return function buildCompilerFromPool (externalSchemas, options) {
+    const externals = JSON.stringify(externalSchemas)
+    const ajvConfig = JSON.stringify(options.customOptions)
```

Comment:
Are we doing those in Fastify directly? I think those will be expensive.

Reply:
Yes, it is so right now.

I would open an issue for it and leave it as-is right now

The solutions I see:

- track the [ajv options](https://github.com/ajv-validator/ajv/blob/master/docs/api.md#options) to write a fast-json-stringify function that serialize them instead of `JSON.stringify`
- apply a new logic like:
  - check the number of keys
  - check iteratively the values (I see only primitive types as values)

Reply:
We can indeed use `fast-json-stringify` at first and bench with it? Also this would only impact cold start. This would be tacled with the use of v8 snapshots